src/EventListener/OidcEndSessionSubscriber.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Drenso\OidcBundle\EventListener;
+
+use Drenso\OidcBundle\Exception\OidcConfigurationException;
+use Drenso\OidcBundle\Exception\OidcConfigurationResolveException;
+use Drenso\OidcBundle\Exception\OidcException;
+use Drenso\OidcBundle\Model\OidcTokens;
+use Drenso\OidcBundle\OidcClientInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\TokenNotFoundException;
+use Symfony\Component\Security\Http\Event\LogoutEvent;
+use Symfony\Component\Security\Http\HttpUtils;
+
+class OidcEndSessionSubscriber implements EventSubscriberInterface
+{
+  public function __construct(
+     private OidcClientInterface $oidcClient,
+     private HttpUtils $httpUtils,
+     private string $logoutTarget
+  ) {
+  }
+
+  /**
+   * @throws OidcConfigurationException
+   * @throws OidcConfigurationResolveException
+   * @throws OidcException
+   */
+  public function onLogout(LogoutEvent $event): void
+  {
+    $token = $event->getToken();
+
+    if (!$token instanceof TokenInterface) {
+      throw new TokenNotFoundException();
+    }
+
+    $oidcTokens = $token->getAttribute('auth_data');
+
+    if (!$oidcTokens instanceof OidcTokens) {
+      throw new OidcException('Invalid token object.');
+    }
```

Comment:
If you check on `instanceof OidcToken`, you can safely use `getAuthData` and be fully typed.

However, the OIDC listeners is not necessarily the only configured listener, so throwing here would be a bit rigoureus. Just returning in case it is not an `OidcToken` should be fine, the only thing missing would be the forward to the external logout link, but at least the user isn't presented with an error.